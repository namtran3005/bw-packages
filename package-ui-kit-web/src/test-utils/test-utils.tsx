import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
