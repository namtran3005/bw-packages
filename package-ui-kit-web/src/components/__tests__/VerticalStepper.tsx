import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import {
  VerticalStepper,
  VerticalStepperStep,
  VerticalStepperSubstep,
} from '../..';

describe('VerticalStepper', () => {
  it('should render Vertical Stepper - 1', () => {
    const { baseElement } = renderWithTestHarness(
      <VerticalStepper>
        <VerticalStepper.Step status="inactive">
          Account
          <VerticalStepper.Substep status="done">
            Password
          </VerticalStepper.Substep>
          <VerticalStepper.Substep status="active">2FA</VerticalStepper.Substep>
        </VerticalStepper.Step>
      </VerticalStepper>
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render Vertical Step', () => {
    const { baseElement } = renderWithTestHarness(
      <VerticalStepperStep status="inactive">
        Account
        <VerticalStepper.Substep status="done">
          Password
        </VerticalStepper.Substep>
        <VerticalStepper.Substep status="active">2FA</VerticalStepper.Substep>
      </VerticalStepperStep>
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render Vertical Stepper - 2', () => {
    const { baseElement } = renderWithTestHarness(
      <VerticalStepperSubstep status="done"></VerticalStepperSubstep>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
