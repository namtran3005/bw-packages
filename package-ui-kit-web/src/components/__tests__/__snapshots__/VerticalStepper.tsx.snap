// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalStepper should render Vertical Step 1`] = `
<body>
  .c1 {
  text-align: left;
}

.c1.c1.c1 {
  font-weight: 500;
  font-size: 1rem;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c3 {
  position: absolute;
  width: 26px;
  height: 26px;
  right: -13px;
  display: inline-block;
  background-color: #FFFFFF;
  box-sizing: border-box;
  border-radius: 50%;
  border-color: #CAC8CB !important;
  border-style: solid;
  border-width: 1px;
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c3::before {
  content: '';
  z-index: 0;
  position: absolute;
  width: 10px;
  height: 10px;
  background: #CAC8CB !important;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c2 {
  cursor: default;
  float: right;
  padding-right: 24px;
  padding-bottom: 16px;
  color: #959196 !important;
}

.c0 {
  display: block;
  clear: both;
  overflow: hidden;
  position: relative;
  overflow: visible;
}

.c0 > div {
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c0.c0.c0 > * {
  white-space: nowrap;
}

.c0 > div {
  max-height: 0px;
  margin: 0px;
  overflow: hidden;
}

<div>
    <div
      class="c0"
    >
      <h6
        class="MuiTypography-root c1 c2 MuiTypography-subtitle1"
      />
      <span
        class="c3"
      />
    </div>
  </div>
</body>
`;

exports[`VerticalStepper should render Vertical Stepper - 1 1`] = `
.c2 {
  text-align: left;
}

.c2.c2.c2 {
  font-weight: 500;
  font-size: 1rem;
  color: #2C232E;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c2.c2.c2:hover {
  background: initial;
}

.c2.c2.c2:active {
  background: initial;
}

.c0 {
  position: relative;
  display: block;
  height: 100%;
}

.c4 {
  position: absolute;
  width: 26px;
  height: 26px;
  right: -13px;
  display: inline-block;
  background-color: #FFFFFF;
  box-sizing: border-box;
  border-radius: 50%;
  border-color: #CAC8CB !important;
  border-style: solid;
  border-width: 1px;
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c4::before {
  content: '';
  z-index: 0;
  position: absolute;
  width: 10px;
  height: 10px;
  background: #CAC8CB !important;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  border-radius: 50%;
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c3 {
  cursor: default;
  float: right;
  padding-right: 24px;
  padding-bottom: 16px;
  color: #959196 !important;
}

.c1 {
  display: block;
  clear: both;
  overflow: hidden;
  position: relative;
  overflow: visible;
}

.c1 > div {
  -webkit-transition: 150ms cubic-bezier(.62,.28,.23,.99);
  transition: 150ms cubic-bezier(.62,.28,.23,.99);
}

.c1.c1.c1 > * {
  white-space: nowrap;
}

.c1 > div {
  max-height: 0px;
  margin: 0px;
  overflow: hidden;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <h6
          class="MuiTypography-root c2 c3 MuiTypography-subtitle1"
        />
        <span
          class="c4"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`VerticalStepper should render Vertical Stepper - 2 1`] = `
<body>
  .c1 {
  text-align: left;
}

.c1.c1.c1 {
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 140%;
  color: #6D6D86;
  font-family: Inter;
  font-style: normal;
  white-space: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  word-wrap: break-word;
}

.c1.c1.c1:hover {
  background: initial;
}

.c1.c1.c1:active {
  background: initial;
}

.c2 {
  padding-bottom: 12px;
  color: #2C232E !important;
}

.c0 {
  cursor: pointer;
  padding-right: 28px;
  display: block;
  float: right;
  clear: both;
  position: relative;
}

<div>
    <div
      class="c0"
    >
      <p
        class="MuiTypography-root c1 c2 MuiTypography-body2"
      />
    </div>
  </div>
</body>
`;
